.github/workflows/package-component.yaml

```diff
@@ -0,0 +1,58 @@
+name: Package on Tag
+
+on:
+  push:
+    tags:
+      - '*/*/*/v*.*.*'
+
+jobs:
+  package:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Extract tag details
+        id: tag_info
+        run: |
+          TAG=${GITHUB_REF#refs/tags/}
+          COMPONENT_TYPE=$(echo $TAG | cut -d/ -f1)
+          COMPONENT_NAME=$(echo $TAG | cut -d/ -f2)
+          VERSION=$(echo $TAG | cut -d/ -f3)
+          
+          echo "component_type=$COMPONENT_TYPE" >> $GITHUB_OUTPUT
+          echo "component_name=$COMPONENT_NAME" >> $GITHUB_OUTPUT
+          echo "version=$VERSION" >> $GITHUB_OUTPUT
+
+      - name: Extract SDK Version
+        id: sdk_version
+        run: |
+          SDK_VERSION=$(grep 'github.com/smithy-security/smithy/sdk' new-components/${{ steps.tag_info.outputs.component_type }}/${{ steps.tag_info.outputs.component_name }}/go.mod | awk '{print $2}')
+          echo "sdk_version=$SDK_VERSION" >> $GITHUB_OUTPUT
+
+      - name: Set up Go
+        uses: actions/setup-go@v5
+        with:
+          go-version: '1.23'
```

Comment:
this should use the go.mod file version

Reply:
I think this is going to be replaced real soon with a pre-compiled and versioned binary - so we can avoid breaking the action while merging new unreleased things in to smithyctl

Reply:
what if we accidentally push a change that breaks smithyctl packaging itself? then the good smithyctl version builds the broken one and pushes it to the registry which causes followup publish steps to fail.

we should build smithyctl itself from the current commit and use that to push itself to the registry. if the command is broken in some way, the next commit will fix it and the workflow will finish successfully without manual intervention.

we are using the go.mod file to set the version automatically in the test workflow: https://github.com/smithy-security/smithy/blob/main/.github/workflows/test.yml#L34

Reply:
Smithyctl is not a component, so it is packaged and distributed differently.

I would prefer to use it pre-compiled, tested and versioned after release to keep things stabler and less prone to break accidentally. 

That said, I'll make the changes and can revisit later if needed

Reply:
Done